### PR TITLE
rubocop: exclude more tap files from the top-level method cop

### PR DESCRIPTION
**Pull request by apainintheneck**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

Basically, this started failing for me on my personal tap because there are some methods defined at the top-level in my Rakefile. That seems acceptable to me and since we're already ignoring all other Ruby files in the tap I figured we might as well just ignore all of them.

I tested it locally and the errors I was getting with my personal tap went away.

Errors: https://github.com/apainintheneck/homebrew-dev-utils/actions/runs/7769885195/job/21189430063?pr=38

### CODE REVIEW

**Comment by apainintheneck** [comment]
I wonder if we should always use `Taps/**/*` instead of `Taps/**/*.rb` to avoid situations like this.

**Review by Bo98 — APPROVED** [review]

**Comment by Bo98** [comment]
> I wonder if we should always use `Taps/**/*` instead of `Taps/**/*.rb` to avoid situations like this.

Probably. I see no reason why it should be limited to `.rb`

**Comment by apainintheneck** [comment]
It turned out that there was only one other use of it. The other ones are all `Taps/*/*/*.rb` which is a little bit more specific.

**Comment by MikeMcQuaid** [comment]
Makes sense, thanks @apainintheneck!
